client/components/signup-site-title/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import LoggedOutForm from 'components/logged-out-form';
+
+import formState from 'lib/form-state';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormButton from 'components/forms/form-button';
+import FormTextInput from 'components/forms/form-text-input';
+
+import { getSiteTitle } from 'state/signup/steps/site-title/selectors';
+
+const SignupSiteTitle = React.createClass( {
+	displayName: 'Signup Site Title',
+
+	propTypes: {
+		onSubmit: React.PropTypes.func.isRequired,
+		siteTitle: React.PropTypes.string.isRequired,
+	},
+
+	componentWillMount() {
+		this.formStateController = new formState.Controller( {
+			fieldNames: [ 'siteTitle' ],
+			validatorFunction: () => {},
+			onNewState: this.setFormState,
+			onError: this.handleFormControllerError,
+			hideFieldErrorsOnChange: true,
+			initialState: {
+				siteTitle: {
+					value: this.props.siteTitle
+				}
+			}
+		} );
+
+		this.setFormState( this.formStateController.getInitialState() );
+	},
+
+	setFormState( state ) {
+		this.setState( { form: state } );
+	},
+
+	handleChangeEvent( event ) {
+		this.formStateController.handleFieldChange( {
+			name: event.target.name,
+			value: event.target.value
+		} );
+	},
+
+	handleFormControllerError( error ) {
+		if ( error ) {
+			throw error;
+		}
```

Comment:
Poked around with the error handler and it seems that if there is no specific validation on the field, the code is not needed at all.